src/view/event-edit-view.js

```diff
@@ -218,60 +213,132 @@ const createEventEditTemplate = (
     this.#allDestinations = allDestinations;
-    this.#availableOffers = availableOffers;
+    this.#allOffers = allOffers;
     this.#handleFormSubmit = onFormSubmit;
     this.#handleFormClose = onFormClose;
-
-    this.form = this.element.querySelector('form');
-    this.form.addEventListener('submit', this.#formSubmitHandler);
-    this.element
-      .querySelector('.event__rollup-btn')
-      .addEventListener('click', this.#formCloseHandler);
+    this._restoreHandlers();
   }
 
   get template() {
     return createEventEditTemplate(
-      this.#event,
-      this.#eventInfo,
+      this._state,
       this.#allDestinations,
-      this.#availableOffers
+      this.#allOffers
     );
   }
 
-  resetForm = () => {
-    this.form.reset();
+  _restoreHandlers() {
+    this.element
+      .querySelector('form')
+      .addEventListener('submit', this.#formSubmitHandler);
+    this.element
+      .querySelector('.event__rollup-btn')
+      .addEventListener('click', this.#formCloseHandler);
+    this.element
+      .querySelector('.event__type-group')
+      .addEventListener('change', this.#typeChangeHandler);
+    this.element
+      .querySelector('.event__input--destination')
+      .addEventListener('change', this.#destinationChangeHandler);
+    this.element
+      .querySelector('.event__field-group--price')
+      .addEventListener('change', this.#priceChangeHandler);
+
+    if (this.element.querySelector('.event__section--offers') !== null) {
+      this.element
+        .querySelector('.event__section--offers')
+        .addEventListener('change', this.#offerChangeHandler);
+    }
+  }
+
+  reset = (event) => {
+    this.updateElement(EventEditView.parseEventToState(event));
   };
 
   #formSubmitHandler = (evt) => {
     evt.preventDefault();
-    this.#handleFormSubmit(this.#event);
+    this.#handleFormSubmit(EventEditView.parseStateToEvent(this._state));
   };
 
   #formCloseHandler = (evt) => {
     evt.preventDefault();
-    //this.form.reset();
-    this.resetForm();
     this.#handleFormClose();
   };
+
+  #typeChangeHandler = (evt) => {
+    evt.preventDefault();
+    if (evt.target.tagName === 'INPUT') {
+      this.updateElement({
+        type: evt.target.value,
+        offers: [],
+      });
+    }
+  };
+
+  #destinationChangeHandler = (evt) => {
+    const name = evt.target.value;
+    const newDestination = this.#allDestinations.find(
+      (destination) => destination.name === name
+    );
+
+    if (newDestination) {
+      this.updateElement({
+        destination: newDestination.id,
+      });
+    }
+  };
+
+  #priceChangeHandler = (evt) => {
+    const newBasePrice = evt.target.value;
+    const PATTERN = /^[\D0]+|\D/g;
+
+    if (newBasePrice && !PATTERN.test(newBasePrice)) {
+      this.updateElement({
+        basePrice: newBasePrice,
+      });
+    }
+  };
+
+  #offerChangeHandler = (evt) => {
+    evt.preventDefault();
+    if (this.element.querySelector('.event__section--offers') !== null) {
```

Comment:
Вообще сравнение с `null` тут излишне.
```suggestion
    if (this.element.querySelector('.event__section--offers')) {
```

Но более важно, а зачем здесь эта проверка. Как можно вызвать этот евент, если нет `.event__section--offers` ?

Reply:
Ага, тут излишне перестраховался, он же в принципе не отрисуется, если офферов нет